pkg/proxy/proxy.go

```diff
@@ -121,9 +135,25 @@ func (p *Proxy) StartProxy() *http.Server {
 
 	// routes
 	wg := router.Group("/apis/toolchain.dev.openshift.com/v1alpha1/workspaces")
+	// Space lister routes
 	wg.GET("/:workspace", handlers.HandleSpaceGetRequest(p.spaceLister))
 	wg.GET("", handlers.HandleSpaceListRequest(p.spaceLister))
 	router.GET(proxyHealthEndpoint, p.health)
+	// SSO routes. Used by web login (oc login -w).
+	// Here is the expected flow for the "oc login -w" command:
+	// 1. "oc login -w <proxy_url>"
+	// 2. oc calls <proxy_url>/.well-known/oauth-authorization-server (wellKnownOauthConfigEndpoint endpoint)
+	// 3. proxy forwards it to <sso_url>/auth/realms/<sso_realm>/.well-known/openid-configuration
+	// 4. oc starts an OAuth flow by opening a browser for <proxy_url>/auth/realms/<realm>/protocol/openid-connect/auth
+	// 5. proxy redirects (the request is not proxied but redirected via 403 See Others response!) the request
+	//    to <sso_url>/auth/realms/<realm>/protocol/openid-connect/auth
+	//    Note: oc uses this hardcoded public (no secret) oauth client name: "openshift-cli-client" which has to exist in SSO to make this flow work.
```

Comment:
I wanted to just aks if we had to do something special for our SSO client, but it looks like that oc has the client name hard-coded, which means that we need to stick with that.

I guess that this means that we needed to configure the dev keycloak to be aware of this client with some special configuration I guess